### PR TITLE
docs: replace codesandbox references with stackblitz

### PR DESCRIPTION
**Pull request by Parsium**
### Proposed behaviour

- Replace all references to Codesandbox with link to [stackblitz starter template](https://stackblitz.com/fork/github/Parsium/carbon-starter)

### Checklist

- [x] Commits follow our style guide
- [x] Related issues linked in commit messages if required
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Additional context

- The Formik demo has not been updated yet as part of this PR: <http://localhost:9001/?path=/docs/documentation-validations--string-validation#formik-overview>
- Currently the code for the starter template used by Stackblitz is hosted externally: [Link to external repo](https://github.com/Parsium/carbon-starter). This ideally should be included in the Carbon repo in the future (see FE-6361).

### Testing instructions

- Check the edited files for any grammatical errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Carbon [![npm](https://img.shields.io/npm/v/carbon-react.svg)](https://www.npmjs.com/package/carbon-react) [![Cypress](https://github.com/Sage/carbon/actions/workflows/cypress.yml/badge.svg)](https://github.com/Sage/carbon/actions/workflows/cypress.yml) [![Playwright](https://github.com/Sage/carbon/actions/workflows/playwright.yml/badge.svg)](https://github.com/Sage/carbon/actions/workflows/playwright.yml)
 
-Carbon is a [React](https://facebook.github.io/react/) component library developed by Sage.
+Carbon is a [React](https://react.dev/) component library developed by Sage.
```
Comment (edleeks87) [inline README.md:3]: good catch 🤦

**Review by edleeks87 — DISMISSED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -85,7 +87,8 @@ body:
   - type: dropdown
     id: os
     attributes:
-      label: What Operating System are you seeing the problem on?
+      label: Relevant OSes
```
Comment (tomdavies73) [inline .github/ISSUE_TEMPLATE/bug_report.yml:90]: **suggestion**: I think the correct pluralization here would be `OSs`, might be wrong tho

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: Feature Request
+name: 🏗️ New feature request
```
Comment (tomdavies73) [inline .github/ISSUE_TEMPLATE/feature_request.yml:1]: **praise**: emojis are on point

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,25 +1,13 @@
...
-If applicable, add screenshots of a codesandbox to help explain your request. You can paste these directly into GitHub.
-
-Please DO NOT share screenshots or the source code of your project.
-
-You can create a codesandbox to show the behaviour before/after this pull request by forking this template https://codesandbox.io/s/carbon-quickstart-j5pb2
-
-If you include a CodeSandbox link, the bot will fork it with the new built version of carbon.
-If you have a commit that includes fixes #XXXX and issue #XXXX has a CodeSandbox link in the body, the bot will fork
-it with the new built version of carbon.
+A clear and concise description of what changes this PR makes. If applicable, include any UI screenshots to help explain your request. If you are a Carbon consumer who is contributing, please DO NOT share any screenshots or source code of your own project.
```
[the remark this answers is not in the document]
Comment (tomdavies73) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: I know this is mentioned elsewhere e.g do not include any sensitive info, however a note that UI from actual products should not be included could be a good addition - especially on a public facing platform

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 5]
```diff
@@ -8,48 +8,48 @@ body:
...
     validations:
       required: true
   - type: textarea
     id: suggested-solution
     attributes:
-      label: Suggested Solution
+      label: Suggested solution
     validations:
-      required: false
+      required: true
```
Comment (tomdavies73) [inline .github/ISSUE_TEMPLATE/feature_request.yml:30]: **praise**: I agree the suggested solution field for a feat request should be required, good spot

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,25 +1,13 @@
...
 -->
 
 ### Current behaviour
 
 <!--
-A clear and concise description of the behaviour before this change.
-
-If applicable, add screenshots. You can paste these directly into GitHub.
+A clear and concise description of the behaviour before this change. If applicable, include any UI screenshots to help explain your request. If you are a Carbon consumer who is contributing, please DO NOT share any screenshots or source code of your own project.
```
Comment (tomdavies73) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: **suggestion**: also mention here that UI screenshots from products should not be included either

**Review by edleeks87 — APPROVED** [review]

**Review by tomdavies73 — APPROVED** [review]

**Pull request merged**